### DISCUSSION TITLE
Batch logs together with counts over period

### DISCUSSION THREAD

**Discussion by 99kennetn** [opcodesio/log-viewer#146] [Ideas]
As i am looking for a lightweight alternative to Sentry, it would be really awsome if errors could be grouped together. Maybe with multiple choices. So if you have something like a user error where you log its id, you could choose between batching them in these two ways.

Error: Error on updating user with id: 3      (grouping by direct title)
or
Error: Error on updating user with id: <int> (grouping by simularities)

in both instances you should be able to scroll through all the occurences of the log stack.

When displayed it would be nice if it showed a count of how many times the batch has been seen in a log or more preferable over all logs, over different timespans like today, yesterday, past week etc....

With these two implementations it would give a nice rundown of what is throwing errors in the system and with quick access to the log files full data in context of the errors around it at the time.

Then down the line stats could be extended into dashboards but then i am getting ahead of myself :D

**Comment by arukompas** [comment]
Hey @99kennetn , welcome and thanks for using the Log Viewer :D 

Love the feedback, although I think it might be too complex/advanced for this open-sourced package :) There are far more important features that should be built first and I can't really compete with the likes of Sentry or Datadog, multi-million dollar companies :) So just being honest here - I don't think this feature will make it into the Log Viewer any time soon.

As a workaround, you can do a lot of filtering by simply searching. For example, I include `[UID XXXX]` on all my logs, where XXXX is the related user ID if I have it. Then it becomes really easy to search for any logs for that user ID. That's just one idea. Don't forget you can now search across multiple files at once :) 

But, since this is a free and open-source project, you're welcome to copy and make any changes you wish. If you feel the community would benefit from the functionality, submit a PR as well!

**Comment by 99kennetn** [comment]
i knew it was a big wish, especially beause of how new this project is, but still wanted to mention it :)

I have never contributed to a project before but this might be a nice starting point, so might look into forking it and seeing what i can do :)

Keep up the good work :)
